dashboards/f1_score_review.py

```diff
@@ -0,0 +1,128 @@
+import numpy as np
+import pandas as pd
+import fortepyan as ff
+import streamlit as st
+import streamlit_pianoroll
+import plotly.graph_objects as go
+from utils import dataset_configuration
+from streamlit.errors import DuplicateWidgetID
+
+from piano_metrics.f1_piano import calculate_f1
+
+
+def plot_f1_time_series(metrics: dict, title: str = "") -> go.Figure:
+    """Create line plot of F1 scores over time."""
+    fig = go.Figure()
+
+    fig.add_trace(
+        go.Scatter(x=metrics["time_points"], y=metrics["f1"], name="F1", line=dict(color="rgb(55, 83, 109)")),
+    )
+
+    fig.add_trace(
+        go.Scatter(
+            x=metrics["time_points"],
+            y=metrics["precision"],
+            name="Precision",
+            line=dict(color="rgb(26, 118, 255)"),
+        )
+    )
+
+    fig.add_trace(
+        go.Scatter(x=metrics["time_points"], y=metrics["recall"], name="Recall", line=dict(color="rgb(98, 182, 149)")),
+    )
+
+    fig.update_layout(
+        title=title,
+        xaxis_title="Time (s)",
+        yaxis_title="Score",
+        showlegend=True,
+    )
+    return fig
+
+
+def main():
+    st.title("F1 Score Analysis Dashboard")
+
+    col1, col2 = st.columns(2)
+    with col1:
+        dataset_1 = dataset_configuration(key="0")
+    with col2:
+        dataset_2 = dataset_configuration(key="1")
+
+    if len(dataset_1) > 0 and len(dataset_2) > 0:
+        piece1 = ff.MidiPiece.from_huggingface(dataset_1[0])
+        piece2 = ff.MidiPiece.from_huggingface(dataset_2[0])
+
+        # Analysis parameters
+        st.header("Analysis Parameters")
+        min_time_unit = st.number_input(
+            "Minimum Time Unit (s)",
+            value=0.01,
+            step=0.001,
+            format="%.3f",
+        )
+        velocity_threshold = st.number_input(
+            "Velocity Threshold",
+            value=30,
+            step=1,
+        )
+        use_pitch_class = st.checkbox(
+            "Use pitch class matching",
+            value=True,
+        )
+
+        # Visualization of pieces
+        st.header("Piece Visualizations")
+        col1, col2 = st.columns(2)
+        with col1:
+            streamlit_pianoroll.from_fortepyan(piece=piece1)
+        with col2:
+            try:
+                streamlit_pianoroll.from_fortepyan(piece=piece2, secondary_piece=piece1)    
+            except DuplicateWidgetID:
+                st.write("Duplicate pianoroll")
```

Comment:
I think it would be better to manage the streamlit keys manually and allow to display the same piece twice on this dashboards

Reply:
This is tricky due to https://github.com/Nospoko/streamlit-pianoroll/issues/8 

I can make it happen by generating a new unique key every time user refreshes the website / changes the piece displayed e.g. using uuid.

Reply:
I just pushed a solution that seems to be working without external sources of uniqueness–let me know if it works for you